File: gear-ecommerce-back/src/main/java/tht/closure/operator/model/entity/VnGeographyDistrict.java
package tht.closure.operator.model.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "t_vn_geography_district")
@Getter
@Setter
public class VnGeographyDistrict extends  AbstractEntity {

    private String name;

    private String prefix;

    @ManyToOne
    @JoinColumn
    private VnGeographyProvince province;

    @OneToMany(mappedBy = "district")
    private List<VnGeographyWard> wards = new ArrayList<>();

}

File: gear-ecommerce-front/src/material/model/VnGeographyDistrict.ts
import {AbstractModel} from "./main/AbstractModel";
import {SubCatalog} from "./SubCatalog";
import {VnGeographyProvince} from "./VnGeographyProvince";
import {VnGeographyWard} from "./VnGeographyWard";

export interface VnGeographyDistrict extends AbstractModel {
    name: string;
    prefix: string;
    province?: VnGeographyProvince;
    wards?: VnGeographyWard[];
}
